core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,7 +64,23 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // Evaluate partitions in parallel. Partitions of each rdd will be cached by the `partitions`
+  // val in `RDD`.
+  private[spark] lazy val parallelPartitionEval: Boolean = {
```

Comment:
The use case I'm interested in is when we load a S3-backed table from Hive. Each partition is one of the RDDs in this union and the partition's splits are planned when getPartitions is called. (We also tracked down [HADOOP-12810](https://issues.apache.org/jira/browse/HADOOP-12810) as part of this work.)